pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go

```diff
@@ -151,16 +150,17 @@ func (e *AzureResourceGraphDatasource) executeQuery(ctx context.Context, query *
 	req.URL.Path = path.Join(req.URL.Path, argQueryProviderName)
 	req.URL.RawQuery = params.Encode()
 
-	ctx, span := tracer.Start(ctx, "azure resource graph query", trace.WithAttributes(
+	ctx, span := tracing.DefaultTracer().Start(ctx, "azure resource graph query", trace.WithAttributes(
 		attribute.String("interpolated_query", query.InterpolatedQuery),
 		attribute.Int64("from", query.TimeRange.From.UnixNano()/int64(time.Millisecond)),
 		attribute.Int64("until", query.TimeRange.To.UnixNano()/int64(time.Millisecond)),
 		attribute.Int64("datasource_id", dsInfo.DatasourceID),
 		attribute.Int64("org_id", dsInfo.OrgID),
-	))
-	defer span.End()
+	),
+	)
 
-	tracer.Inject(ctx, req.Header, span)
```

Comment:
Note on removing this should be fine as long as you're using the httpclient package/provider/client (which I'm pretty sure you do here) to make HTTP requests since it does this automatically for you, see https://github.com/grafana/grafana/blob/1b4c2fc94892fb2aaefbc9f377b5f8cec22066d1/pkg/infra/httpclient/httpclientprovider/tracing_middleware.go#L37 and https://github.com/grafana/grafana-plugin-sdk-go/blob/dd41f8d709969a5efd5375833f9754cecf9176bc/backend/httpclient/tracing_middleware.go#L45

Just something to keep in mind since without this the receiving HTTP server won't receive trace id identifier in HTTP header and be able to add spans to an existing trace id - this obviously requires both grafana, the plugin and the HTTP server to send traces to the same storage.

Reply:
Thank you for the info!